### ISSUE TITLE
Slack notification plugin error proxy

### ISSUE THREAD

**Issue by jbravo** [jenkinsci/slack-plugin#102]
Jenkins ver. 1.604
Slack Plugin ver. 1.7

This is the log:

```
...
Using proxy authentication (user=XXX)
May 26, 2015 6:15:19 PM INFO org.apache.commons.httpclient.auth.AuthChallengeProcessor selectAuthScheme
ntlm authentication scheme selected
May 26, 2015 6:15:19 PM SEVERE org.apache.commons.httpclient.HttpMethodDirector executeConnect
Credentials cannot be used for NTLM authentication: org.apache.commons.httpclient.UsernamePasswordCredentials
org.apache.commons.httpclient.auth.InvalidCredentialsException: Credentials cannot be used for NTLM authentication: org.apache.commons.httpclient.UsernamePasswordCredentials
    at org.apache.commons.httpclient.auth.NTLMScheme.authenticate(NTLMScheme.java:332)
    at org.apache.commons.httpclient.HttpMethodDirector.authenticateProxy(HttpMethodDirector.java:320)
    at org.apache.commons.httpclient.HttpMethodDirector.executeConnect(HttpMethodDirector.java:491)
    at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:391)
    at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:171)
    at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
    at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:323)
    at jenkins.plugins.slack.StandardSlackService.publish(StandardSlackService.java:66)
    at jenkins.plugins.slack.ActiveNotifier.notifyStart(ActiveNotifier.java:65)
    at jenkins.plugins.slack.ActiveNotifier.started(ActiveNotifier.java:53)
    at jenkins.plugins.slack.SlackNotifier$SlackJobProperty.prebuild(SlackNotifier.java:280)
    at hudson.model.AbstractBuild$AbstractBuildExecution.preBuild(AbstractBuild.java:816)
    at hudson.model.AbstractBuild$AbstractBuildExecution.preBuild(AbstractBuild.java:811)
    at hudson.model.AbstractBuild$AbstractBuildExecution.preBuild(AbstractBuild.java:807)
    at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:533)
    at hudson.model.Run.execute(Run.java:1751)
    at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
    at hudson.model.ResourceController.execute(ResourceController.java:89)
    at hudson.model.Executor.run(Executor.java:240)

May 26, 2015 6:15:19 PM INFO org.apache.commons.httpclient.HttpMethodDirector processProxyAuthChallenge
Failure authenticating with NTLM <any realm>@example.com:80
...
```

**Comment by samrocketman** [comment]
I don't have a decent way of testing this as I don't have a proxy available.  I leave this up to you or another community member to fix/build/test.  The only testing I'll be able to perform is to ensure the plugin properly works without a proxy configuration when merging the fix.  I don't actively develop this plugin.  I only maintain pull requests and plugins.  Contributions are welcome!

**Comment by nevermindr** [comment]
I was experiencing the same problems 
This got fixed after I've rewritten httpClient.POST code with apache httpclient4.3.2
Submitted Pull Request https://github.com/jenkinsci/slack-plugin/pull/154

@jbravo would be great if you could test this also in your environment

**Comment by nevermindr** [comment]
submitted https://github.com/jenkinsci/slack-plugin/pull/156

**Comment by mattumotu** [comment]
I am also having this issue, I believe my HTTP Proxy Configuration is correct (as I can update and download plugins) but when I try to use the stack plugin I get errors the following error:

Using proxy authentication (user=<domain>\<user>)
Aug 04, 2016 1:55:39 PM INFO org.apache.commons.httpclient.auth.AuthChallengeProcessor selectAuthScheme
ntlm authentication scheme selected
Aug 04, 2016 1:55:39 PM SEVERE org.apache.commons.httpclient.HttpMethodDirector executeConnect
Credentials cannot be used for NTLM authentication: org.apache.commons.httpclient.UsernamePasswordCredentials
org.apache.commons.httpclient.auth.InvalidCredentialsException: Credentials cannot be used for NTLM authentication: org.apache.commons.httpclient.UsernamePasswordCredentials
	at org.apache.commons.httpclient.auth.NTLMScheme.authenticate(NTLMScheme.java:332)
	at org.apache.commons.httpclient.HttpMethodDirector.authenticateProxy(HttpMethodDirector.java:320)
	at org.apache.commons.httpclient.HttpMethodDirector.executeConnect(HttpMethodDirector.java:491)
	at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:391)
	at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:171)
	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:323)
	at jenkins.plugins.slack.StandardSlackService.publish(StandardSlackService.java:72)
	at jenkins.plugins.slack.ActiveNotifier.completed(ActiveNotifier.java:114)
	at jenkins.plugins.slack.SlackListener.onCompleted(SlackListener.java:27)
	at jenkins.plugins.slack.SlackListener.onCompleted(SlackListener.java:15)
	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:202)
	at hudson.model.Run.execute(Run.java:1786)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)

Aug 04, 2016 1:55:39 PM INFO org.apache.commons.httpclient.HttpMethodDirector processProxyAuthChallenge
Failure authenticating with NTLM <any realm>@<example.com>:80
Aug 04, 2016 1:55:39 PM WARNING jenkins.plugins.slack.StandardSlackService publish
Error posting to Slack
java.io.IOException: Stream closed
	at java.io.BufferedInputStream.getBufIfOpen(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	at org.apache.commons.httpclient.ContentLengthInputStream.read(ContentLengthInputStream.java:170)
	at java.io.FilterInputStream.read(Unknown Source)
	at org.apache.commons.httpclient.AutoCloseInputStream.read(AutoCloseInputStream.java:108)
	at java.io.FilterInputStream.read(Unknown Source)
	at org.apache.commons.httpclient.AutoCloseInputStream.read(AutoCloseInputStream.java:127)
	at org.apache.commons.httpclient.HttpMethodBase.getResponseBody(HttpMethodBase.java:690)
	at org.apache.commons.httpclient.HttpMethodBase.getResponseBodyAsString(HttpMethodBase.java:803)
	at jenkins.plugins.slack.StandardSlackService.publish(StandardSlackService.java:73)
	at jenkins.plugins.slack.ActiveNotifier.completed(ActiveNotifier.java:114)
	at jenkins.plugins.slack.SlackListener.onCompleted(SlackListener.java:27)
	at jenkins.plugins.slack.SlackListener.onCompleted(SlackListener.java:15)
	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:202)
	at hudson.model.Run.execute(Run.java:1786)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)

**Comment by nevermindr** [comment]
@samrocketman this is fixed as per #296
